components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -421,18 +422,23 @@ private String handleUserConsent(HttpServletRequest request, String consent, OAu
                 builder.setCode(authzRespDTO.getAuthorizationCode());
                 addUserAttributesToCache(sessionDataCacheEntry, authzRespDTO.getAuthorizationCode(), authzRespDTO.getCodeId());
             }
-            if (StringUtils.isNotBlank(authzRespDTO.getAccessToken())){
+            if (StringUtils.isNotBlank(authzRespDTO.getAccessToken()) &&
+                !OIDCConstants.OIDCCoreConstants.ID_TOKEN.equals(oauth2Params.getResponseType())) {
```

Comment:
Would be better to make this check more strict, as this would allow any response type other than 'id_token'. If we can specifically check whether the response type String is equal to a combination of 'token' and 'id_token', in security aspects it will be more tight and favored.